siliconcompiler/core.py

```diff
@@ -1852,6 +1852,16 @@ def _check_flowgraph(self, flow=None):
                 self.logger.error(f'{step} is not defined in the {flow} flowgraph')
                 error = True
 
+        from_nodes = self._get_execution_entry_nodes(flow)
+        to_nodes = self._get_execution_exit_nodes(flow)
+        reachable_nodes = set(self._reachable_flowgraph_nodes(flow, set(from_nodes)))
+        unreachable_nodes = set(to_nodes).difference(reachable_nodes)
+        if unreachable_nodes:
+            unreachable_nodes_formated = list(
+                map(lambda node: f'{node[0] + node[1]}', unreachable_nodes))
+            self.logger.error(f'These final nodes can not be reached: {unreachable_nodes_formated}')
```

Comment:
Maybe update to say: `These final nodes in {flow} can not be reached`? I'll leave that up to you